crates/iota-data-ingestion-core/README.md

```diff
@@ -0,0 +1,352 @@
+- `remote_fetch()` method fetches checkpoints from a remote store. It supports the Full Node REST API and various object-store interfaces, including [Amazon S3](https://docs.rs/object_store/latest/object_store/aws/index.html), [Google Cloud Storage](https://docs.rs/object_store/latest/object_store/gcp/index.html), and [WebDAV](https://docs.rs/object_store/latest/object_store/http/index.html). Based on the configuration, it can also create a hybrid client combining an object store and a REST API client. It fetches checkpoints in batches, starting from the `current_checkpoint_number`, and retrieves data until the `batch_size` is reached. While iterating through the fetched checkpoints, it checks for capacity limitations. If the capacity is exceeded, it stops fetching and returns the collected checkpoints.
+
+### Progress Store
+
+The `ProgressStore` plays a crucial role in tracking the progress of checkpoint synchronization for each task. This ensures that the Worker can resume synchronization from the last successfully processed checkpoint after an Indexer restart. The framework offers two built-in implementations:
+
+- `ShimProgressStore`: A simple, in-memory progress store primarily used for unit testing. It does not persist progress across restarts.
+- `FileProgressStore`: A persistent progress store that uses a file to track each task's synchronization state.
+
+Custom progress stores can be created by implementing the `ProgressStore` trait.
+
+```rust
+#[async_trait]
+pub trait ProgressStore: Send {
+    async fn load(&mut self, task_name: String) -> Result<CheckpointSequenceNumber>;
+    async fn save(
+        &mut self,
+        task_name: String,
+        checkpoint_number: CheckpointSequenceNumber,
+    ) -> Result<()>;
+}
+```
+
+Instead of directly using the custom `ProgressStore` implementation, the `IndexerExecutor` interacts with it through a wrapper:
+
+```rust
+pub type ExecutorProgress = HashMap<String, CheckpointSequenceNumber>;
+
+pub struct ProgressStoreWrapper<P> {
+    // Our custom implemented ProgressStore
+    progress_store: P,
+    // An internal Map holding the TaskName and the CheckpointSequenceNumber
+    pending_state: ExecutorProgress,
+}
+```
+
+This wrapper implements the `ProgressStore` trait. Internally, it retrieves progress data from the underlying `ProgressStore` implementation and populates its own internal cache. This cached data is then used to efficiently calculate the minimum watermark across all tracked tasks.
+
+### WorkerPool
```

Comment:
I suggest adding links to all methods mentioned.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/4570/commits/c58ed8cab8960cc11588dd32e7a540ae8ebe4c3c#diff-31a167158fe911c55387e5f89c28edd3fd006968b761a550e15148aee28192cbR202-R222)